third_party/amd/lib/TritonAMDGPUTransforms/BlockPingpong.cpp

```diff
@@ -772,6 +843,389 @@ LogicalResult Pingponger::sliceDot(OpBuilder &builder, Location loc,
   return success();
 }
 
+LogicalResult Pingponger::genAsyncCopySlices(OpBuilder &builder) {
+  if (asyncCopyOps.empty())
+    return success();
+
+  auto &loopBody = forOp.getRegion().front();
+  auto yieldOp = cast<scf::YieldOp>(loopBody.getTerminator());
+
+  for (auto asyncCopy : asyncCopyOps) {
+    MLIRContext *ctx = asyncCopy->getContext();
+    auto srcPointers = asyncCopy.getSrc();
+    auto subView = cast<triton::gpu::MemDescSubviewOp>(
+        asyncCopy.getResult().getDefiningOp());
+    auto subViewEncoding = subView.getType().getEncoding();
+
+    DenseMap<int64_t, triton::gpu::MemDescSubviewOp> subViews;
+    for (auto user : subView->getUsers()) {
+      if (auto subView = dyn_cast<triton::gpu::MemDescSubviewOp>(user)) {
+        if (auto attr = subView->getAttrOfType<mlir::IntegerAttr>(
+                Pingponger::sliceAttrName)) {
+          auto sliceIdx = attr.getValue().getSExtValue();
+          subViews.insert({sliceIdx, subView});
+
+          if (!newAsyncGroups.contains(sliceIdx)) {
+            newAsyncGroups.insert({sliceIdx, {}});
+          }
+        }
+      }
+    }
+
+    if (subViews.empty()) {
+      auto commit =
+          getSingleUserOf<ttg::AsyncCommitGroupOp>(asyncCopy.getToken());
+      preservedAsyncCommits.insert(commit->getResult());
+      continue;
+    }
+
+    // infer the sliced shape
+    triton::gpu::MemDescSubviewOp subViewSlice = subViews[0];
+    auto origShape = subView.getType().getShape();
+    auto slicedShape = subViewSlice.getType().getShape();
```

Comment:
Just double checking, is the subViewSlice coming from
```
      Value newOrigSmem = builder.create<ttg::MemDescSubviewOp>(
          origMemDesc.getLoc(), subviewDescType, origMemDesc, offsetsVal);
```

Reply:
We slice 2 subviews for each tile - i.e., 1) one which proceeds `local_load` immediately (usually comes from an arg for `scf::ForOp` and 2) the subview from a previous loop iteration (used by either `local_store` or `async_copy_local_to_global`. The former is called `newSmem`; the later is `origMemDesc`